docs/development.rst

```diff
@@ -0,0 +1,283 @@
+
+.. _`Development`:
+
+Development
+===========
+
+This section only needs to be read by developers of the pywbemtools package.
+People that want to make a fix or develop some extension, and people that
+want to test the project are also considered developers for the purpose of
+this section.
+
+
+.. _`Repository`:
+
+Repository
+----------
+
+The repository for pywbemtools is on GitHub:
+
+https://github.com/pywbem/pywbemtools
+
+
+.. _`Setting up the development environment`:
+
+Setting up the development environment
+--------------------------------------
+
+It is recommended to use Linux as the development environment for pywbem.
+OS-X should work as well, but Windows requires a number of manual setup steps.
+
+1. Clone the Git repo of this project and switch to its working directory:
+
+   .. code-block:: bash
+
+        $ git clone git@github.com:pywbem/pywbemtools.git
+        $ cd pywbemtools
+
+2. It is recommended that you set up a `virtual Python environment`_.
+   Have the virtual Python environment active for all remaining steps.
+
+3. Install pywbem and its prerequisites for installing and running it
+   as described in :ref:`Installation`.
+   This will install Python packages into the active Python environment,
+   and OS-level packages.
+
+4. On Windows, perform the setup steps described in
+   :ref:`Manual setup on Windows`. On Linux and OS-X, the corresponding
+   setup is performed automatically as part of the next step.
+
+5. Install the prerequsites for pywbem development.
+   This will install Python packages into the active Python environment,
+   and OS-level packages:
+
+   .. code-block:: bash
+
+        $ make develop
+
+6. This project uses Make to do things in the currently active Python
+   environment. The command:
+
+   .. code-block:: bash
+
+        $ make
+
+   displays a list of valid Make targets and a short description of what each
+   target does.
+
+.. _virtual Python environment: https://docs.python-guide.org/en/latest/dev/virtualenvs/
+
+
+.. _`Manual setup on Windows`:
```

Comment:
This section basically has nothing to manually set up. If there is nothing, the section can as well be removed.

Reply:
I suggest to ignore this comment for now, and possibly come back to it in a final docs review.

Reply:
Completely rewrote